app/controllers/webhooks_controller.rb

```diff
@@ -1,33 +1,42 @@
 # frozen_string_literal: true
 require 'samson/integration'
 
-class WebhooksController < ApplicationController
+class WebhooksController < ResourceController
   include CurrentProject
 
   before_action :authorize_resource!
 
-  def index
-  end
-
   def create
     webhook = current_project.webhooks.build(webhook_params)
-    if webhook.save
-      redirect_to project_webhooks_path(current_project), notice: "Webhook created!"
-    else
-      current_project.reload
-      flash[:alert] = "Error saving webhook: #{webhook.errors.full_messages.join(", ")}"
-      render :index
+    respond_to do |format|
+      format.html do
+        if webhook.save
+          redirect_to project_webhooks_path(current_project), notice: "Created!"
+        else
+          flash[:alert] = "Error saving webhook: #{webhook.errors.full_messages.join(", ")}"
+          render :index
+        end
+      end
+      format.json do
+        webhook.save!
+        render_as_json 'webhooks', webhook
+      end
     end
   end
 
-  def destroy
-    webhook = current_project.webhooks.find(params[:id])
-    webhook.soft_delete!(validate: false)
+  private
 
-    redirect_to project_webhooks_path(current_project)
+  def search_resources
+    @project.webhooks
   end
 
-  private
+  def resources_path
+    project_webhooks_path(current_project)
+  end
+
+  def resource_params
+    super.permit(:branch, :source)
```

Comment:
I think when an added `.merge(project: current_project)` you can delete all of `def create`